components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -116,35 +117,68 @@ void BlockchainRegistry::GetAllTokens(const std::string& chain_id,
   std::move(callback).Run(std::move(tokens_copy));
 }
 
-void BlockchainRegistry::GetBuyTokens(mojom::OnRampProvider provider,
-                                      const std::string& chain_id,
-                                      GetBuyTokensCallback callback) {
+std::vector<brave_wallet::mojom::BlockchainTokenPtr>
+BlockchainRegistry::GetBuyTokens(
+    const std::vector<mojom::OnRampProvider>& providers,
+    const std::string& chain_id) {
   std::vector<brave_wallet::mojom::BlockchainTokenPtr> blockchain_buy_tokens;
-  const std::vector<mojom::BlockchainToken>* buy_tokens = nullptr;
-  if (provider == mojom::OnRampProvider::kWyre)
-    buy_tokens = &GetWyreBuyTokens();
-  else if (provider == mojom::OnRampProvider::kRamp)
-    buy_tokens = &GetRampBuyTokens();
-  else if (provider == mojom::OnRampProvider::kSardine)
-    buy_tokens = &GetSardineBuyTokens();
-  else if (provider == mojom::OnRampProvider::kTransak)
-    buy_tokens = &GetTransakBuyTokens();
-
-  if (buy_tokens == nullptr) {
-    std::move(callback).Run(std::move(blockchain_buy_tokens));
-    return;
+  std::vector<mojom::BlockchainToken> buy_tokens;
+  if (std::find(providers.begin(), providers.end(),
+                mojom::OnRampProvider::kWyre) != providers.end()) {
+    std::vector<mojom::BlockchainToken> wyre_buy_tokens = GetWyreBuyTokens();
+    buy_tokens.insert(buy_tokens.begin(), wyre_buy_tokens.begin(),
+                      wyre_buy_tokens.end());
```

Comment:
wyre has been removed https://github.com/brave/brave-browser/issues/27678 so don't think we need this.